src/rgw/rgw_acl.h

```diff
@@ -214,20 +217,25 @@ struct ACLReferer {
       perm(perm) {
   }
 
-  bool is_match(std::string http_referer) const {
-    if (http_referer == url_spec) {
+  bool is_match(boost::string_ref http_referer) const {
+    const auto http_host = get_http_host(http_referer);
+    if (!http_host) {
+      return false;
+    }
+
+    boost::string_ref host_spec(url_spec);
```

Comment:
if using `operator==`, it will return error. So I use `->compare`  to compare `http_host` with `url_spec` insteaded.